presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushProjectionThroughExchange.java

```diff
@@ -104,6 +106,20 @@ public Result apply(ProjectNode project, Captures captures, Context context)
                 projections.put(exchange.getPartitioningScheme().getHashColumn().get(), exchange.getPartitioningScheme().getHashColumn().get().toSymbolReference());
                 inputs.add(exchange.getPartitioningScheme().getHashColumn().get());
             }
+
+            if (exchange.getOrderingScheme().isPresent()) {
+                // need to retain ordering columns for the exchange
```

Comment:
Would you consider adding a test to trigger the new code paths in `PushProjectionThroughExchange` rule?

Reply:
I think it is tested: `TestPushProjectionThroughExchange#testOrderingColumnsArePreserved`